src/main/java/com/commercetools/sync/services/CustomObjectService.java

```diff
@@ -0,0 +1,74 @@
+package com.commercetools.sync.services;
+
+
+import com.commercetools.sync.commons.models.NonResolvedReferencesCustomObject;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.customobjects.CustomObject;
+import io.sphere.sdk.customobjects.CustomObjectDraft;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+
+public interface CustomObjectService {
+
+    /**
+     * Given a custom object key, this method fetches a custom object that matches this given key in the CTP project
+     * defined in a potentially injected {@link SphereClient}. If there is no matching custom object an empty
+     * {@link Optional} will be returned in the returned future. A mapping of the key to the id of the fetched
+     * custom object is persisted in an in -memory map.
+     *
+     * @param key the key of the custom object to fetch.
+     * @return {@link CompletionStage}&lt;{@link Optional}&gt; in which the result of it's completion contains an
+     * {@link Optional} that contains the matching {@link CustomObject} if exists, otherwise empty.
+     */
+    @Nonnull
+    CompletionStage<Optional<CustomObject<NonResolvedReferencesCustomObject>>> fetchCustomObject(
```

Comment:
`fetch` is enough

Reply:
c031550e391deee0adff7c5299d099062963158a